### PR TITLE
lib: implementation of `i128`

### LINKED ISSUE #3107: simple arithmetic does not work for `i128`
`u128` works as expected with this small example:
```
 > ./build/bin/fz -e "say (u128.one + u128.one)"
2
```
`i128` does not:
```
 > ./build/bin/fz -e "say (i128.one + i128.one)"

$MODULE/i128.fz:28:8: error 1: Used abstract features 'num.wrap_around.wrapped_on_neg', 'num.wrap_around.overflow_on_add', 'num.wrap_around.underflow_on_add', 'num.wrap_around.prefix -°', 'numeric.infix %', 'numeric.as_u8' are not implemented by 'i128'
public i128(public hi i64, public lo u64) : num.wrap_around, has_interval is
-------^^^^
Feature 'i128' instantiated at $MODULE/i128.fz:45:28:
  public fixed type.max => i128 0x_7fff_ffff_ffff_ffff 0x_ffff_ffff_ffff_ffff
---------------------------^^^^
inherits or declares abstract feature 'num.wrap_around.wrapped_on_neg' declared at $MODULE/num/wrap_around.fz:36:10:
  public wrapped_on_neg bool => abstract

```

### PR DESCRIPTION
**Pull request by simonvonhackewitz**
fix #3107

### CODE REVIEW

File: tests/basicIntegers/basicIntegers.fz [hunk 1]
```diff
@@ -39,20 +39,24 @@ basicIntegers =>
...
-  test_u64 (a, b i64) unit => test u64 a.cast_to_u64           b.cast_to_u64
-  test_u32 (a, b i64) unit => test u32 a.low32bits            b.low32bits
-  test_u16 (a, b i64) unit => test u16 a.low16bits            b.low16bits
-  test_u8  (a, b i64) unit => test u8  a.low8bits             b.low8bits
+  test_i128 (a, b i64) unit => test i128 a.as_i128               b.as_i128
+  test_i64  (a, b i64) unit => test i64  a                       b
+  test_i32  (a, b i64) unit => test i32  a.low32bits.cast_to_i32 b.low32bits.cast_to_i32
+  test_i16  (a, b i64) unit => test i16  a.low16bits.cast_to_i16 b.low16bits.cast_to_i16
+  test_i8   (a, b i64) unit => test i8   a.low8bits.cast_to_i8   b.low8bits.cast_to_i8
+  test_u128 (a, b i64) unit => test u128 a.as_i128.cast_to_u128  b.as_i128.cast_to_u128
```
Comment (michaellilltokiwa) [inline tests/basicIntegers/basicIntegers.fz:47]: `as_i128.cast_to_u128` seems a bit weird? Is as_u128 missing?

File: tests/i128/i128_test.fz [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test i128_test
+#
+# -----------------------------------------------------------------------
+
+i128_test =>
+
+  i128.hx_out =>
```
Comment (michaellilltokiwa) [inline tests/i128/i128_test.fz:26]: *_out seem unused remove them?

File: tests/i128/i128_test.fz [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+  test_mul (i128 13 23) i128.one (i128 13 23) ""
+  test_mul (i128 3 2) -(i128 2) -(i128 6 4) "-110680464442257309700"
+
+  test_eq ((i128 2)**(i128 126)) (i_bits 0x4000_0000_0000_0000 0)
+
+  test_div i128.one i128.one (i128 1) "1"
+  test_div i128.max i128.one (i128.max) ""
+  test_div (i128.min + i128.one) -i128.one (i128.max) ""
+  test_div (i128.min) i128.one (i128.min) ""
```
Comment (michaellilltokiwa) [inline tests/i128/i128_test.fz:143]: div by zero case is missing.

File: tests/i128/i128_test.fz [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  test_eq -(i128.min + i128.one) (i128.max)
+
+  test_sub i128.one i128.one i128.zero "0"
+  test_sub -i128.one i128.one (-i128 2) "-2"
+  test_sub -i128.one -(i128 2) (i128 1) "1"
+
+  test_mul i128.one i128.one (i128 1) "1"
+  test_mul i128.min i128.one i128.min ""
+  test_mul i128.min i128.zero i128.zero "0"
+  test_mul i128.max i128.zero i128.zero "0"
```
Comment (michaellilltokiwa) [inline tests/i128/i128_test.fz:128]: i128.max *° 128.max is missing

File: tests/i128/i128_test.fz [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  # 0..01_0..01 + 0..01_0..01 = 0..010_0..010
+  test_add (i128 1 1) (i128 1 1) (i128 2 2) "36893488147419103234"
+  # 53 + (-11) = 42
+  test_add (i128 53) (-i128 11) (i128 42) "42"
+
+  test_eq -(i128.min + i128.one) (i128.max)
+
+  test_sub i128.one i128.one i128.zero "0"
+  test_sub -i128.one i128.one (-i128 2) "-2"
+  test_sub -i128.one -(i128 2) (i128 1) "1"
```
Comment (michaellilltokiwa) [inline tests/i128/i128_test.fz:123]: test_sub with 128.min/128.max is missing

**Review by michaellilltokiwa — APPROVED** [review]
Nice!

File: tests/basicIntegers/basicIntegers.fz (hunk 1 above)
Comment (simonvonhackewitz) [reply tests/basicIntegers/basicIntegers.fz:47]: In this test for all the unsigned types the bit representation of the signed input type is cast to an unsigned type. E.g.`u64`
```
  test_u64  (a, b i64) unit => test u64  a.cast_to_u64           b.cast_to_u64
```

So `.as_i128` just makes up for the input type being of smaller size than the type in the test. Then `cast_to_u128` takes the bit representation and treats it as an `u128`, like in the other tests.

**Pull request merged**